File: 70-climbing-stairs/70-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        vector<int> steps(n+1,0);
        steps[0] = 1;
        steps[1] = 1;
        
        if(n == 0 || n == 1) { // 0 steps away or 1 step away
            return steps[n];
        }
        
        for(int i = 2; i <= n ; i++) { // i is the steps away from target n
            steps[i] = steps[i-1] + steps[i-2];
        }
        
        return steps[n];
    }
};

File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        prev2 = 1
        prev = 1
        
        for i in range(2,n+1):
            curSteps = prev2 + prev
            prev2 = prev
            prev = curSteps
        
        return prev
    
# Time Complexity: O(n)
# Space Complexity: O(1)